### PR TITLE
Optimizing connect to bootNodes failure code & Remove a superfluous l…

### PR DESCRIPTION
**Pull request by af913337456**
一些修改如下，谢谢：
1. 优化了一下 DHT 连接 boot 节点失败判断条件的代码，避免了多次使用 len 读取一样的 cfg.BootNodes；
2. 增加 SelectUtxos 方法的可读性;
3. 删除了一次多余的 clearExpiredLocks 调用。

### CODE REVIEW

**Comment by af913337456** [comment]
@fxsjy 你好，请问 Travis 的这个基于 Mac 环境下的错误怎么处理，配置文件我并没有做过修改。
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

**Comment by af913337456** [comment]
Linux 环境下的部署检测，Travis 是通过了的

File: utxo/utxo.go [hunk 1]
```diff
@@ -569,14 +569,14 @@ func (uv *UtxoVM) GenerateRootTx(js []byte) (*pb.Transaction, error) {
...
 func (uv *UtxoVM) SelectUtxos(fromAddr string, fromPubKey string, totalNeed *big.Int, needLock, excludeUnconfirmed bool) ([]*pb.TxInput, [][]byte, *big.Int, error) {
-	uv.clearExpiredLocks()
-	curLedgerHeight := uv.ledger.GetMeta().TrunkHeight
-	willLockKeys := make([][]byte, 0)
-	utxoTotal := big.NewInt(0)
-	foundEnough := false
-	// 先从cache里找找，不够再从leveldb找,因为leveldb prefix scan比较慢
-	cacheKeys := map[string]bool{}
-	txInputs := []*pb.TxInput{}
+	var (
```
Comment (yucaowang) [inline utxo/utxo.go]: It's not our code style to declare and initialize local variables using "var". instead, we use short variable declarations.

Comment (af913337456) [reply utxo/utxo.go]: Means that your code style is not allowed to be modified, is it? @yucaowang

Comment (yucaowang) [reply utxo/utxo.go]: I think it's better to keep consistent with the code repo if there isn't any strong reason.

Comment (af913337456) [reply utxo/utxo.go]: Understand, in addition, the above Mac problem is also caused by the code style? @yucaowang

Comment (yucaowang) [reply utxo/utxo.go]: I'm not sure, but I have triggered the CI build again, let's see if it's a occasional problems.

Comment (af913337456) [reply utxo/utxo.go]: Thanks alot.

Comment (yucaowang) [reply utxo/utxo.go]: CI build passed, please go ahead.

Comment (af913337456) [reply utxo/utxo.go]: @yucaowang looks like it's all passed.

Comment (af913337456) [reply utxo/utxo.go]: @yucaowang The code style has been restored. Is it up to your requirements now?

**Pull request merged**